hack/generator/pkg/codegen/pipeline/add_status_conditions.go

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+)
+
+const AddStatusConditionsStageID = "addStatusConditions"
+
+func AddStatusConditions(idFactory astmodel.IdentifierFactory) Stage {
+	return MakeLegacyStage(
+		AddStatusConditionsStageID,
+		"Adds conditions to all status types and implements genruntime.Conditioner on all resources",
+		func(ctx context.Context, defs astmodel.Types) (astmodel.Types, error) {
+			result := make(astmodel.Types)
+
+			statusDefs := astmodel.FindStatusTypes(defs)
+			for _, def := range statusDefs {
+				statusType, err := addConditions(def.Type())
+				if err != nil {
+					return nil, errors.Wrapf(err, "couldn't add Conditions condition to status %q", def.Name())
+				}
+
+				result.Add(def.WithType(statusType))
+			}
+
+			resourceDefs := astmodel.FindResourceTypes(defs)
+			for _, def := range resourceDefs {
+				resourceType := def.Type().(*astmodel.ResourceType)
+
+				conditionerImpl, err := astmodel.NewConditionerInterfaceImpl(idFactory, resourceType)
+				if err != nil {
+					return nil, errors.Wrapf(err, "couldn't create conditioner impl for %q", def.Name())
+				}
+				resourceType = resourceType.WithInterface(conditionerImpl)
+
+				result.Add(def.WithType(resourceType))
+			}
+			remaining := defs.Except(resourceDefs).Except(statusDefs)
+			// Add everything that's left
+			for _, def := range remaining {
+				result.Add(def)
+			}
+
+			return result, nil
+		})
+}
+
+// addConditions adds a "Conditions" property to the status type
+func addConditions(statusType astmodel.Type) (astmodel.Type, error) {
+	visitor := astmodel.TypeVisitorBuilder{
+		VisitObjectType: addConditionsToObject,
+	}.Build()
+
+	return visitor.Visit(statusType, nil)
+}
```

Comment:
Use `PropertyInjector` to do the heavy lifting

Reply:
That is nice!

The only problem is that I want to make sure to error if we're trying to add a property that already exists. It seems like maybe `PropertyInjector` should have that behavior by default too? Otherwise we'll be silently overwriting properties.

I'll delete my code here and update `PropertyInjector` to do that.